### PR TITLE
[tests-only][full-ci] Improve Then step in apiSharingNotificationsToShares suites

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of the `When` steps.

**Suites covered:**
- `apiSharingNotificationsToShares`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by SwikritiT — COMMENTED** [review]
LGTM

**Review by grgprarup — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature [hunk 1]
```diff
@@ -114,31 +122,38 @@ Feature: Display notifications when receiving a share
...
     And the administrator adds user "David" to group "grp1" using the provisioning API
-    Then user "Brian" should have 0 notifications
+    Then the HTTP status code of responses on all endpoints should be "200"
+    And the OCS status code of responses on each endpoint should be "100, 200, 100" respectively
+    And user "Brian" should have 0 notifications
     And user "Carol" should have 0 notifications
     And user "David" should have 0 notifications
 
   Scenario: when auto-accepting is enabled no notifications are sent
-    Given parameter "shareapi_auto_accept_share" of app "core" has been set to "yes"
```
Comment (SagarGi) [inline tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature:126]: is auto-accept set to "yes"  by default as you are removing this step. @amrita-shrestha

**Review by SagarGi — COMMENTED** [review]
other looks good to me :+1:

File: tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature [hunk 2]
```diff
@@ -114,31 +122,38 @@ Feature: Display notifications when receiving a share
 
   @skipOnLDAP
   Scenario: share to group does not send notifications to existing and new members for a share created after auto-accept is enabled
-    Given parameter "shareapi_auto_accept_share" of app "core" has been set to "yes"
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature:117]: why is this step deleted?

Comment (kiranparajuli589) [inline tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature:117]: the scenario title suggests this should be enabled before sharing, why removing?

File: tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature (hunk 1 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature:126]: why removing?

File: tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature (hunk 2 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature:117]: Doesn't shareapi_auto_accept_share enable by default?

Comment (phil-davis) [reply tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature:117]: > Doesn't shareapi_auto_accept_share enable by default?

yes, it does. But I think the scenario is more understandable when it is explicitly mentioned that auto-accept is enabled.

Comment (amrita-shrestha) [reply tests/acceptance/features/apiSharingNotificationsToShares/sharingNotifications.feature:117]: ok

**Review by phil-davis — APPROVED** [review]

**Review by kiranparajuli589 — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Pull request merged**